pkg/utils/utils.go

```diff
@@ -52,6 +56,25 @@ func AreSelectorsSame(lhs, rhs map[string]string) bool {
 	return true
 }
 
+// EnsureLinuxNodeSelector takes given selector map and returns a selector map with linux node selector added into it.
+// If there is already a node type selector and is different from "linux" then it is kept intact but warning is logged.
+// See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#interlude-built-in-node-labels
+func EnsureLinuxNodeSelector(selectors map[string]string) map[string]string {
+	if selectors == nil {
+		return map[string]string{OsNodeLabel: LinuxValue}
+	}
+	if os, ok := selectors[OsNodeLabel]; ok {
+		if os == LinuxValue {
+			return selectors
+		}
+		// Can this not be "linux"?
+		logrus.Warnf("Unexpected node selector value: %s=%s", OsNodeLabel, os)
+	} else {
```

Comment:
i would expect that we always make sure the os node label is `LinuxValue` based on the description of the card and function name... is this not the case?

Reply:
This is probably the case when user explicitly set the node selector (which by default is not set). I can make it override the value anyway, just wanted to make sure this is what we really want to do (i.e. ignore and override users settings).

Reply:
@jcantrill do you know the answer to this?

Reply:
Fixed.